.github/workflows/test-and-publish.yaml

```diff
@@ -1,29 +1,34 @@
-name: 'Testing Suite'
+name: 'Test and Publish Suite'
 
 on:
   push:
```

Comment:
```suggestion
  push:
    tags:
      - '*'
```

Reply:
hmm.... This makes it so the tests only run on tags no? By default just setting 
```
push:
```
should also run for tags, but doing:
```
push:
  tags:
    - '*'
```
Only runs for tagged pushes. I've changed it so that the publish step is only run during tags, so we should be able to allow the test to run all the time, for any push. 

Let me know if that makes sense.

Reply:
I think it's good to run the tests for every push on `main` or pull requests, but not for any push on any branch. It's a waste to run test for commits with WIP changes that expects to fail. 

Also, it duplicates CI runs for both `pull_request` and `push` events.
![](https://github.com/Distributive-Network/PythonMonkey/assets/16057139/e7b336c9-9875-41c4-b2c7-6eb93135211d)

Reply:
I buy that, Okay. I've changed it to run on pushes to main and on pushes with tags. PR's are still there and will always run. This should dedupe the runs on PR's and will prevent running every single time for branches.